--- a/storage/innobase/include/fts0types.ic
+++ b/storage/innobase/include/fts0types.ic
@@ -116,14 +116,32 @@
 fts_is_charset_cjk(
 	const CHARSET_INFO*	cs)
 {
-	return cs == &my_charset_gb2312_chinese_ci
+	return 0
+#ifdef HAVE_CHARSET_GB2312
+    	|| cs == &my_charset_gb2312_chinese_ci
+#endif
+#ifdef HAVE_CHARSET_GBK
 		|| cs == &my_charset_gbk_chinese_ci
+#endif
+#ifdef HAVE_CHARSET_BIG5
 		|| cs == &my_charset_big5_chinese_ci
+#endif
+#ifdef HAVE_CHARSET_UJIS
 		|| cs == &my_charset_ujis_japanese_ci
+#endif
+#ifdef HAVE_CHARSET_SJIS
 		|| cs == &my_charset_sjis_japanese_ci
+#endif
+#ifdef HAVE_CHARSET_CP932
 		|| cs == &my_charset_cp932_japanese_ci
+#endif
+#ifdef HAVE_CHARSET_EUCJPMS
 		|| cs == &my_charset_eucjpms_japanese_ci
-		|| cs == &my_charset_euckr_korean_ci;
+#endif
+#ifdef HAVE_CHARSET_EUCKR
+		|| cs == &my_charset_euckr_korean_ci
+#endif
+		;
 }

 /** Select the FTS auxiliary index for the given character by range.