Index: mariadb-10.3.2/CMakeLists.txt
===================================================================
--- mariadb-10.3.2.orig/CMakeLists.txt
+++ mariadb-10.3.2/CMakeLists.txt
@@ -436,6 +436,7 @@ IF(NOT CMAKE_CROSSCOMPILING)
   # minimal target to build only binaries for export
   ADD_CUSTOM_TARGET(import_executables DEPENDS ${EXPORTED})
   EXPORT(TARGETS ${EXPORTED} FILE ${CMAKE_BINARY_DIR}/import_executables.cmake)
+  ADD_CUSTOM_TARGET(IMPORT_EXECUTABLES DEPENDS comp_err comp_sql factorial gen_lex_hash)
 ENDIF()
 
 CONFIGURE_FILE(config.h.cmake   ${CMAKE_BINARY_DIR}/include/my_config.h)
